### DISCUSSION TITLE
Only return unique values in dataview query?

### DISCUSSION THREAD

**Discussion by noellecdv0** [mdelobelle/metadatamenu#261] [Q&A]
I love that you can use a dataview query to populate a field's options, but the one issue I'm running into is that when I wrote my query:

````dv.pages('#Novel/BookTitle/Character').map(p => p.role)````

... It outputted every character's role value rather than a condensed list of the unique roles to choose from. Screenshot:

![image](https://user-images.githubusercontent.com/98427340/211176990-b38e32bc-65e9-416f-a001-d9a0258bfa96.png)

Is there any way to adjust the DV query to return only unique values, and if not, it would be amazing if it could be added as an option in the future!

**Comment by mdelobelle** [comment]
Hi! you can use this method to have unique values in your select:
`[...new Set(dv.pages('#Novel/BookTitle/Character').map(p=>p.role))]`

**Comment by noellecdv0** [comment]
> Hi! you can use this method to have unique values in your select: `[...new Set(dv.pages('#Novel/BookTitle/Character').map(p=>p.role))]`

That doesn't seem to work, unfortunately. Just returned an empty list.

![image](https://user-images.githubusercontent.com/98427340/211180547-35d64f3c-0d7e-4ffc-9dfc-760acdbff4a8.png)

**Comment by mdelobelle** [comment]
Do you see any error in the console?

Can you try alternatively the dataviewjs data array method `dv.pages('#Novel/BookTitle/Character').map(p=>p.role).distinct()` ?
